x/supplier/client/cli/query_proof.go

```diff
@@ -0,0 +1,78 @@
+package cli
+
+import (
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/spf13/cobra"
+
+	"github.com/pokt-network/poktroll/x/supplier/types"
+)
+
+func CmdListProof() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list-proofs",
+		Short: "list all proofs",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			pageReq, err := client.ReadPageRequest(cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			queryClient := types.NewQueryClient(clientCtx)
+
+			params := &types.QueryAllProofsRequest{
+				Pagination: pageReq,
+			}
+
+			res, err := queryClient.AllProofs(cmd.Context(), params)
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintProto(res)
+		},
+	}
+
+	flags.AddPaginationFlagsToCmd(cmd, cmd.Use)
+	flags.AddQueryFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func CmdShowProof() *cobra.Command {
```

Comment:
Should there a `// TODO_TECHDEBT` for reacting to updates in the index type or something?